changelog.d/16740.bugfix

```diff
@@ -0,0 +1,3 @@
+Write signing keys with file mode 0640
+
+Contributed by Fabian Klemp
```

Comment:
CI will complain about the formatting here. I've also tweaked the prose a little.

```suggestion
Fix a long-standing bug where the signing keys generated by Synapse were world-readable. Contributed by Fabian Klemp.
```